### PR TITLE
Concatenate the day's survey responses

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
Blocked by #5

Adds a "cron" job to run a method every day. This method downloads all responses from the day and combines them into a new csv file. 

Everything is broken up into helper methods so that the behavior can be tweaked as desired.

### TODO
- [ ] Ensure that all files concatenated come from correct folder? We can't guarantee the folder structure so perhaps it's better to leave things as is.
- [x] Run concatenation on startup to avoid weird behavior upon crash

### CODE REVIEW

File: server/util.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+const { parse } = require('json2csv')
+
+const config = require('../server.config.json')
+
+/**
+ * Async/await wrapper for listing all objects in an S3 bucket
+ */
+const listObjectsInBucket = async (s3, bucket) => {
+  return new Promise((resolve, reject) => {
+    s3.listObjects({ Bucket: bucket }, function (err, data) {
```
Comment (daniel-heppner-ibigroup) [inline server/util.js]: Does this method not have a `.promise()`?

File: server/util.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    await Promise.all(
+      filteredFilesInBucket.map(async (file) => {
+        const fileContents = await getS3ObjectContents(s3, bucket, file.Key)
+        return csv().fromString(fileContents)
+      })
+    )
+  ).flat()
+
+  // This object must first be initialized with all keys from all responses
+  // TODO: should these keys come from the client config instead? That way they wouldn't
```
Comment (daniel-heppner-ibigroup) [inline server/util.js:75]: Hmmm. Yeah, that would definitely be an improvement but it feels like the client should be putting these in the CSV header itself, which then could be read in here. I think that can be a separate PR and this is fine for now.

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Great work! I hate that we're converting to and from CSV so much but I guess it's okay so that we can always have CSV files in S3.

File: server/util.js (hunk 2 above)
Comment (miles-grant-ibigroup) [reply server/util.js:75]: That's a good point! Agreed and will open a new issue.

File: server/util.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply server/util.js]: Wow I did not know about that! Nice catch thanks!

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM and seems to work.
